src/content/en/2021/cdn.md

```diff
@@ -17,4 +17,420 @@ featured_stat_3: TODO
+**Caveats and Disclaimers**
+
+As with any observational study, there are limits to the scope and impact that can be measured. The statistics gathered on CDN usage for the Web Almanac are focused more on applicable technologies in use and not intended to measure performance or effectiveness of a specific CDN vendor. While this ensures that we are not biased towards any CDN vendor, it also means that these are more generalized results.
+
+These are the limits to our testing methodology:
+
+**Simulated network latency:** We use a dedicated network connection that synthetically shapes traffic.
+
+**Single geographic location:** Tests are run from a single datacenter and cannot test the geographic distribution of many CDN vendors.
+
+**Cache effectiveness:** Each CDN uses proprietary technology and many, for security reasons, do not expose cache performance.
+
+**Localization and internationalization: **Just like geographic distribution, the effects of language and geo-specific domains are also opaque to these tests.
+
+**CDN detection:** This is primarily done through DNS resolution and HTTP headers. Most CDNs use a DNS CNAME to map a user to an optimal datacenter. However, some CDNs use AnyCast IPs or direct A+AAAA responses from a delegated domain which hide the DNS chain. In other cases, websites use multiple CDNs to balance between vendors, which is hidden from the single-request pass of WebPageTest. 
+
+All of this limits the effectiveness in the measurements.
+
+Most importantly, these results reflect the utilization of specific features (Example: TLS, HTTP/2 etc.,) but do not reflect actual impact. YouTube is more popular than “www.example.com” yet both will appear as equal value when comparing utilization.
+
+With this in mind, here are a few statistics that were intentionally not measured in the context of a CDN:
+
+
+
+1. Time To First Byte(TTFB)
+2. CDN Round Trip Time
+3. Core Web Vitals
+4. Cache Hit v/s Cache Miss performance etc.,
+
+**CDN Adoption**
+
+The contents in a web page can be divided into 3 parts, namely:
+
+
+
+1. Base HTML Page (Ex: [www.example.com](www.example.com))
+2. Embedded First Party Content (Ex: images.example.com, css.example.com etc.,)
+3. 3rd Party Content (Ex: Google Analytics, Advertisements etc.,)
+
+From their inception, CDNs have been the go-to solution for delivering embedded content like: images, stylesheets, javascripts, fonts, etc.  This kind of  content doesn't change frequently, making it a good candidate for caching on a CDN’s proxy servers.
+
+With the evolution of CDN technology an expressway was set up on the internet for non-cacheable assets.  This means HTMLs and APIs can now be delivered reliably and faster, compared to the standard TCP connection.
+
+
+
+{{ figure_markup(
+  image="CDN Usage vs Hosted Resources.png",
+  caption="CDN Usage vs Hosted Resources.",
+  description="Bar chart showing the CDN Usage for domains serving HTML, sub-domains serving static assets, third party domains",
```

Comment:
It'd be helpful to be more descriptive in all figures `description` fields. If you were to try to describe the contents of the figure to someone over the phone, what would you say?

For example, in this chart I'd say something like this:
```suggestion
  description="Bar chart of CDN usage versus hosted resource split by origin and CDN. For HTML content, 73% of requests are from the origin and 27% from CDNs. For sub-domain content, 62% origin and 38% CDN. And for third-party content, 37% origin and 63% CDN.",
```